gobblin-core/src/main/java/gobblin/writer/FsDataWriter.java

```diff
@@ -75,12 +82,33 @@ public FsDataWriter(State properties, String fileName, int numBranches, int bran
       this.fs = FileSystem.get(URI.create(uri), conf);
     }
 
-    // initialize staging/output dir
+    // Initialize staging/output directory
     this.stagingFile = new Path(WriterUtils.getWriterStagingDir(properties, numBranches, branchId), fileName);
     this.outputFile = new Path(WriterUtils.getWriterOutputDir(properties, numBranches, branchId), fileName);
-    this.properties.setProp(
-        ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FINAL_OUTPUT_PATH, branchId),
-        this.outputFile.toString());
+    this.properties
+        .setProp(ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FINAL_OUTPUT_PATH, branchId),
+            this.outputFile.toString());
+
+    this.bufferSize = Integer.parseInt(properties.getProp(
+        ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_BUFFER_SIZE, numBranches, branchId),
+        ConfigurationKeys.DEFAULT_BUFFER_SIZE));
+
+    this.replicationFactor = properties.getPropAsShort(ForkOperatorUtils
+        .getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_REPLICATION_FACTOR, numBranches, branchId),
+        this.fs.getDefaultReplication(this.outputFile));
+
+    this.blockSize = properties.getPropAsLong(
+        ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_BLOCK_SIZE, numBranches, branchId),
+        this.fs.getDefaultBlockSize(this.outputFile));
+
+    this.permission = new FsPermission(properties.getPropAsShort(
+        ForkOperatorUtils.getPropertyNameForBranch(ConfigurationKeys.WRITER_FILE_PERMISSIONS, numBranches, branchId),
+        FsPermission.getDefault().toShort()));
+
+    Optional<String> group = Optional.fromNullable(properties.getProp(ConfigurationKeys.WRITER_GROUP_NAME));
+    if (group.isPresent()) {
+      this.fs.setOwner(this.stagingFile, this.fs.getFileStatus(this.stagingFile).getOwner(), group.get());
```

Comment:
I'm not sure if this will work, because technically `this.stagingFile` doesn't exist yet. It only gets created in the `createDatumWriter` method of `AvroHdfsDataWriter`.